### DISCUSSION TITLE
Web3Modal in vanillaJS

### DISCUSSION THREAD

**Discussion by wendy-bear** [WalletConnect/walletconnect-monorepo#1931] [Developer Support]
Hello)
I'm new to frontend in web3modal theme and I have a question about using web3modal in vanilla js.
We have an application completely written in vanilla js without any webpack-type builders and so on. I need to add a web3modal to it.
I connect all the logic according to the example: 
https://github.com/WalletConnect/web3modal-vanilla-js-example
and for some reason everything works fine)

But the following questions still remain:

1) Is it possible to test sending transactions in a test network like Goerli or Sepolia? So far, in my application, transactions are successfully sent only on the Ethereum network.
2) Is it possible to somehow add a test network to the provider's settings? Here:
const providerOptions = {
    walletconnect: {      
      package: WalletConnectProvider,
      options: {
        infuraId: "someID",       
      },
      rpc: {
        5: "https://goerli.infura.io/v3/",
      },
    },
  };

3) Is it planned to accidentally stop supporting v1 for vanilla js? My apps are too simple to use webpack like version 2( 
Or will I still have to use the builder and version 2 in the near future?

**Comment by xzilja** [comment]
Hey @wendy-bear looks like you stumbled upon one of the older repositories for web3modal version that is no longer supported. We have a new one http://web3modal.com which has `@web3modal/html` package https://docs.walletconnect.com/2.0/web3modal/html-js/installation

There is no cdn/url link at the moment, but what you can do is take package above (with all associated libs as well) and run it through something like https://vitejs.dev or https://webpack.js.org to produce a bundle file that you can then use within your app.

**Reply by wendy-bear** [reply]
Hey @0xAsimetriq )
So I need to use webpack (or vite) and import deps in this way:
import { SafeConnector } from '@gnosis.pm/safe-apps-wagmi'
import { configureChains, createClient } from '@wagmi/core'
import { goerli, mainnet } from '@wagmi/core/chains'
import { EthereumClient, modalConnectors, walletConnectProvider } from '@web3modal/ethereum'
import { Web3Modal } from '@web3modal/html' and so on.

and then I can use usual vanillaJS for further development?
And what about test networks? If I'll import { goerli, mainnet } from '@wagmi/core/chains' can I test a sending transactions in Goerli/Sepolia networks? Because I heard that WalletConnect works only in main net like Ethereum

**Reply by xzilja** [reply]
In addition to above you also need to set all of these on `window` so you can access them via for example `window.SafeConnector`, `window.wagmi`
So it might make sense to `import * as WagmiChains from "@wagmi/core/chains"` for you to have access to all of them.
You can use whatever chain you want, just need to make sure your test wallet supports it and you provide rpc for it. I'd suggest looking into https://wagmi.sh/core/getting-started documentation for further details.

wagnmi is essentially a wrapper on top of ethersjs that we integrated into web3modal, so all ethereum related tasks happen there and web3modal is ui on top of this.

**Reply by wendy-bear** [reply]
Thank you) I'm going to read the all documentation rigth now

**Reply by wendy-bear** [reply]
Hello again)
I figured out almost everything, but I do not understand how to initiate the provider and the contract.
Previously, I initiated a web3 entity via const web3 = new Web3(provider) from V1. 
And my contract was initiated via window.web3 = new window.Web3(provider || window.ethereum);
      CONTRACT = new window.web3.eth.Contract(ABI, contractAddress);
For example, I want my provider to be infura. Here information about infura like provider
https://wagmi.sh/core/providers/infura
However, I don't understand how, based on this provider, to initiate a contract like const daiContract = new ethers.Contract(daiAddress, daiAbi, provider);
Should i use ethers lib or can i use web3js?
If I can, how do I access the contract? For vanillaJS version

**Reply by xzilja** [reply]
You don't need to initiate it, you can use wagmi actions like readContract / writeContract etc... for example https://wagmi.sh/core/actions/readContract you just need to provide correct address for the chain + abi (for typescript auto complete)

**Reply by xzilja** [reply]
If everything is working, do you mind sharing your setup?  Perhaps I could add it to our build pipline quickly so we provide cdn script 👍

**Reply by wendy-bear** [reply]
Thanks for your answers)
I installed vite and used imports, without using cdn) It was easy)

But I want to understand the web3modal structure for the vanilla version. Do I understand correctly that first I need to import all dependencies, then, following the example here https://github.com/WalletConnect/web3modal/tree/V2/examples/html, add a button widget to the index and other settings to the js file?
I then use the actions from here https://wagmi.sh/core/actions/ to access contract methods, sign messages and send transactions?
So I don't need to initialize the provider at all?
But how can I listen for network or account changes? Previously listeners were like in these examples provider.on("accountsChanged", (accounts) => {
     setTimeout(() => {
       window.location.reload();
     }, 2000);
     //fetchAccountData();
   });
And how to do it in version 2?

**Reply by xzilja** [reply]
All signer / provider actions like these are exposed by wagmi i.e. for above, there is `watchAccount` action. Wagmi is much simpler to use as well. web3modal is just a ui built on top of it with main functionality around connecting user's wallet and showing some account data. You can use either pre-built button or your own. Again, I suggest you deep dive into both web3modal and wagmi docs to understand all of this better. There are also examples available for html i.e https://github.com/WalletConnect/web3modal/tree/V2/examples/html

**Reply by CalculatingZero** [reply]
Would it be possible to initiate web3 like this: ```const web3 = new Web3(ethereumClient);```? I would like to continue to use the web3js lib if possible.

I also tested this with the following script tags and it seems to connect fine, but the V1 modal no longer pops up. I tried the latest version of the V1 modal and that didn't work either.

```
<script src="https://unpkg.com/@walletconnect/ethereum-provider@2.4.7/dist/index.umd.js"></script>
<script src="https://unpkg.com/web3modal@1.9.5/dist/index.js"></script>

let n = {main: 'eip155:1', goerli: 'eip155:5'};
 const projectId = config.projectId;

const web3Modal = new Web3Modal.default({projectId: projectId,standaloneChains: [n[config.network]],});
  let provider = await web3Modal.connect();
  const web3 = new Web3(provider);
  const accounts = await web3.eth.getAccounts();
  const selectedAccount = accounts[0];
  const marketplaceContract = config.contract;
```

**Reply by oleksiivinogradov** [reply]
> Hey @wendy-bear looks like you stumbled upon one of the older repositories for web3modal version that is no longer supported. We have a new one http://web3modal.com which has `@web3modal/html` package https://docs.walletconnect.com/2.0/web3modal/html-js/installation
> 
> There is no cdn/url link at the moment, but what you can do is take package above (with all associated libs as well) and run it through something like https://vitejs.dev or https://webpack.js.org to produce a bundle file that you can then use within your app.

I sure that a lot of apps (include Unity games) works with pure JS and will have issues 100%
As result all of them must remove walletconnect and leave metamask support only (forever)
You guys push us to migrate but even can't support vanila JS (or provide full working guide for webpack)

Why you ruin own business?
